frontend/src/api/k8sUtils.ts

```diff
@@ -45,27 +58,58 @@ export const groupVersionKind = (model: K8sModelCommon): K8sGroupVersionKind =>
   kind: model.kind,
 });
 
+// RFC 7386 (JSON Merge Patch):
+// - Setting an object property to null removes it from the target.
+// - Arrays are replaced as a whole; individual elements are not patched.
+// - This function replaces undefined with null in object properties only, not in arrays.
+// - See also: https://www.npmjs.com/package/json-merge-patch
+export const deepReplaceUndefinedWithNull: {
+  <T>(obj: T): T;
+  (obj: unknown): unknown;
+} = (obj: unknown): unknown => {
+  if (Array.isArray(obj)) {
+    // Do not replace undefined with null in arrays; recurse into objects inside arrays only
+    return obj.map((v) =>
+      typeof v === 'object' && v !== null && !Array.isArray(v)
+        ? deepReplaceUndefinedWithNull(v)
+        : v,
+    );
+  }
+  if (obj && typeof obj === 'object') {
+    const result: Record<string, unknown> = {};
+    for (const [key, value] of Object.entries(obj)) {
+      result[key] = value === undefined ? null : deepReplaceUndefinedWithNull(value);
+    }
+    return result;
+  }
+  return obj;
+};
+
 export const k8sMergePatchResource = <
   TResource extends K8sResourceCommon,
+  TPatch extends K8sMergePatch<TResource> = K8sMergePatch<TResource>,
   TUpdatedResource extends TResource = TResource,
 >({
   model,
   resource,
   queryOptions = {},
   fetchOptions = {},
-}: K8sResourceUpdateOptions<TResource>): Promise<TUpdatedResource> => {
-  if (!resource.metadata?.name) {
+}: K8sResourceUpdateOptions<TPatch>): Promise<TUpdatedResource> => {
+  if (!resource.metadata.name) {
     return Promise.reject(new Error('Resource payload name not specified'));
   }
 
+  // Convert all undefined values to null for merge patch
+  const resourceWithNulls = deepReplaceUndefinedWithNull(resource);
```

Comment:
this may be undesired behavior for _all_ merge patches we do. if the user expects the standard behavior then this would break that. I would suggest just fixing it at the hardware profile level